sympy/algebras/quaternion.py

```diff
@@ -875,6 +950,18 @@ def _ln(self):
 
         return Quaternion(a, b, c, d)
 
+    def _eval_subs(self, *args):
+        elements = [i.subs(*args) for i in self.args]
+        norm = self._norm
+        try:
+            _check_norm(elements, norm)
+        except ValueError:
+            norm = None
+            warnings.warn("Substitution renders pre-computed norm value "
```

Comment:
This seems like it could make for a hard to track error -- shouldn't this be a fatal error?

Reply:
I think you're right, I'll change that!

Reply:
Alright, changed that too, and now the `_eval_subs` also acts on `_norm` .